File: src/features/workouts/__tests__/restTimer.test.ts
import { formatClock, isResting, restRemainingSeconds } from '@/features/workouts/restTimer';

describe('restRemainingSeconds', () => {
  it('is 0 when there is no rest end', () => {
    expect(restRemainingSeconds(null, 1000)).toBe(0);
  });
  it('rounds up remaining time and floors at 0', () => {
    expect(restRemainingSeconds(10_500, 0)).toBe(11);
    expect(restRemainingSeconds(500, 1000)).toBe(0);
  });
  it('isResting reflects remaining > 0', () => {
    expect(isResting(5000, 0)).toBe(true);
    expect(isResting(0, 1000)).toBe(false);
  });
});

describe('formatClock', () => {
  it('formats seconds as M:SS', () => {
    expect(formatClock(0)).toBe('0:00');
    expect(formatClock(9)).toBe('0:09');
    expect(formatClock(75)).toBe('1:15');
    expect(formatClock(600)).toBe('10:00');
  });
});

File: android-native/app/src/test/java/com/fitnessrpg/app/domain/workouts/RestTimerTest.kt
package com.fitnessrpg.app.domain.workouts

import org.junit.Assert.assertEquals
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class RestTimerTest {
    @Test
    fun `is 0 when there is no rest end`() {
        assertEquals(0, restRemainingSeconds(null, 1000L))
    }

    @Test
    fun `rounds up remaining time and floors at 0`() {
        assertEquals(11, restRemainingSeconds(10_500L, 0L))
        assertEquals(0, restRemainingSeconds(500L, 1000L))
    }

    @Test
    fun `isResting reflects remaining greater than zero`() {
        assertTrue(isResting(5000L, 0L))
        assertFalse(isResting(0L, 1000L))
    }

    @Test
    fun `formats seconds as minutes and seconds`() {
        assertEquals("0:00", formatClock(0))
        assertEquals("0:09", formatClock(9))
        assertEquals("1:15", formatClock(75))
        assertEquals("10:00", formatClock(600))
    }
}
